src/pages/CoinsPage/CoinsPage.tsx

```diff
@@ -0,0 +1,98 @@
+import styles from './CoinsPage.module.scss';
+import React, { useEffect } from 'react';
+import { Outlet, useLocation, useSearchParams } from 'react-router-dom';
+import Search from '../../components/Search';
+import Footer from '../../components/Footer';
+import Pagination from '../../components/Pagination';
+import {
+  COINS_LIMIT,
+  SEARCH_VALUE_IN_LOCAL_STORAGE,
+} from '../../constants/constants';
+import Loader from '../../components/Loader';
+import CoinsList from '../../components/CoinsList';
+import { coinsApi } from '../../store';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import FlyoutMenu from '../../components/FlyoutMenu';
+import { useAppDispatch, useAppSelector } from '../../hooks/reduxTypedHooks';
+import Header from '../../components/Header';
+import { setCoinsFromPage } from '../../store/slices/coinsOnPageSlice';
+import { getTotalPagesNumber } from '../../utils/getTotalPagesNumber ';
+
+const { app, main, coins, aside } = styles;
+
+const CoinsPage: React.FC = () => {
+  const [searchValue, setSearchValue] = useLocalStorage(
+    SEARCH_VALUE_IN_LOCAL_STORAGE,
+    ''
+  );
+
+  const location = useLocation();
+  const [searchParams, setSearchParams] = useSearchParams(location.search);
+  const page = parseInt(searchParams.get('page') || '1', 10);
```

Comment:
So, I've noticed 2 places where you perform this same action; why don't you **create a custom hook** for determining the current page?